### DISCUSSION TITLE
Generating 3D surface of the gaussian process surpack surrogate with UI

### DISCUSSION THREAD

**Discussion by MandeepS1n** [snl-dakota/dakota#17] [Q&A] [answered]
Hey, 
I am trying to plot the 3D surface of the gaussian Process surfack surrogate model generated on an LHS sampling study. 
I know that a surface plot of the multidim_parameter_study can be made in DakotaUI, but I couldn't find out how to make a surface plot of the gaussian surrogate model in UI. I tried to plot using the export_approx_points_file of the surrogate model but it did not work.
Is there a way to plot it with DakotaUI or does it need to be done with python. Any help is really appreciated.

Regards,  
Mandeep Singh

**Comment by jadamstephens** [comment, marked as answer]
Mandeep,

Set up a Dakota study that performs a multidim_parameter_study on the Gaussian process. It'll look something like the input file in [this example](https://github.com/snl-dakota/dakota-examples/tree/master/official/surrogates/dace). The main difference will be that the 'EvalSurrogate' method should be changed to a multidim_parameter_study. You'll also need to add a few keywords to the environment block to write out the data to be plotted.

```
environment
  method_pointer = 'EvalSurrogate'
  results_output hdf5
  tabular_data
```
The surrogate evaluations will be written to both dakota_results.h5 (hdf5 format) and dakota_tabular.dat (whitespace delimited text). You can use the plotting tools in the GUI to plot them, or you can do it on your own using Python, Matlab, Excel, etc.

Adam

**Reply by eridgway90** [reply]
To add to Adam's answer, here is some documentation on how to use Dakota GUI's plotting capabilities to visualize your data, though you may prefer to use an external plotting tool regardless.

[Plot Trace Creator dialog](https://dakota.sandia.gov/sites/default/files/docs/latest_release/user-html/usingdakotagui/chartreuse/PlotTraceCreator.html)
[Getting Plot Data from a Dakota Tabular Data File](https://dakota.sandia.gov/sites/default/files/docs/latest_release/user-html/usingdakotagui/chartreuse/PlotTraceCreator.html#getting-plot-data-from-a-dakota-tabular-data-file)
[Surface 3D plots](https://dakota.sandia.gov/sites/default/files/docs/latest_release/user-html/usingdakotagui/chartreuse/PlotTraceCreator.html#surface-3d-plot)

**Reply by MandeepS1n** [reply]
Thanks a lot. This was really helpful. 

I had one more question. Is it also possible to plot the sample points onto the contour plot. I tried plotting the contour, then editing it and adding the sample points to it, but it did not work.

**Reply by eridgway90** [reply]
At this time, the Dakota GUI does not allow for manually adding sample points directly to a trace. However, you can add additional traces to the same canvas (i.e. the same X/Y/Z coordinates axis) using the [Plot Manager dialog](https://dakota.sandia.gov/sites/default/files/docs/latest_release/user-html/usingdakotagui/chartreuse/PlotManager.html). This would be the preferred solution if your original sample points are already in a Dakota-friendly format (such as another tabular data file).

Hope this helps.
